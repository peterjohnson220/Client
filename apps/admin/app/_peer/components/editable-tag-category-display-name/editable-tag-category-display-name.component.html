<span title="Click to Edit Display Name" class="editable" *ngIf="!isEditing" (click)="toggleEdit()">
  {{value}} <span class="editable-icon"> <fa-icon icon="pencil"> </fa-icon> </span>
</span>
<ng-container *ngIf="isEditing">
  <form [formGroup]="editDisplayNameForm" (ngSubmit)="save()">
    <div class="input-group">
      <input id="editable-category-name-input"
             type="text"
             class="form-control"
             autocomplete="off"
             formControlName="tempValue"
             pfFocus
             [pfValidatable]="!isSaving"
             [minLength]="3">
      <div (click)="save()" class="input-group-append" [ngClass]="{'disabled': !saveButtonEnabled}">
          <span class="input-group-text">
            <fa-icon icon="check-circle" class="text-success"></fa-icon>
          </span>
      </div>
      <div (click)="cancel()" class="input-group-append" [ngClass]="{'disabled': isSaving}">
          <span class="input-group-text">
            <fa-icon icon="times"></fa-icon>
          </span>
      </div>

    </div>

    <ng-container *ngIf="tempValueControl.errors">
      <div *ngIf="tempValueControl.errors.required" class="invalid-feedback">Display Name is required.</div>
      <div *ngIf="tempValueControl.errors.error" class="invalid-feedback">{{tempValueControl.errors.error}}</div>
    </ng-container>
  </form>
</ng-container>
