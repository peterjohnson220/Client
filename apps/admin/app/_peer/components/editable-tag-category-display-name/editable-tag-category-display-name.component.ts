import {Component, EventEmitter, Input, OnDestroy, OnInit, Output} from '@angular/core';
import {FormBuilder, FormGroup} from '@angular/forms';

import { Observable, Subscription } from 'rxjs';

import { PfValidators } from 'libs/forms/validators';

@Component({
  selector: 'pf-editable-tag-category-display-name',
  templateUrl: './editable-tag-category-display-name.component.html',
  styleUrls: ['./editable-tag-category-display-name.component.scss']
})
export class EditableTagCategoryDisplayNameComponent implements OnInit, OnDestroy {
  @Input() value: string;
  @Input() saving$: Observable<boolean>;
  @Input() errorSaving$: Observable<boolean>;
  @Output() onSave = new EventEmitter();

  savingSubscription: Subscription;
  editDisplayNameForm: FormGroup;
  isEditing: boolean;
  isSaving: boolean;

  constructor(private fb: FormBuilder) {
    this.createForm();
  }

  get tempValueControl() { return this.editDisplayNameForm.get('tempValue'); }

  get isDirty(): boolean { return this.value !== this.tempValueControl.value; }

  get saveButtonEnabled(): boolean {
    return this.isDirty && !this.isSaving && this.tempValueControl.valid;
  }

  createForm(): void {
    this.editDisplayNameForm = this.fb.group({
      'tempValue': [{value: false, disabled: true}, [PfValidators.required, PfValidators.minLengthTrimWhitespace(3)]]
    });
  }

  save(): void {
    if (!this.saveButtonEnabled) {
      return;
    }

    this.onSave.emit(this.tempValueControl.value);
  }

  cancel(): void {
    if (this.isSaving) {
      return;
    }

    this.reset();
  }

  reset(): void {
    this.isEditing = false;
    this.tempValueControl.setValue(this.value);
    this.tempValueControl.enable();
  }

  toggleEdit(): void {
    this.isEditing = true;
    this.tempValueControl.setValue(this.value);
    this.tempValueControl.enable();
  }

  // Events
  ngOnInit(): void {
    this.savingSubscription = this.saving$.subscribe(saving => {
      this.isSaving = saving;

      if (!this.isEditing) {
        return;
      }

      if (saving) {
        this.tempValueControl.disable();
      } else {
        this.reset();
      }
    });
  }

  ngOnDestroy(): void {
    this.savingSubscription.unsubscribe();
  }
}
