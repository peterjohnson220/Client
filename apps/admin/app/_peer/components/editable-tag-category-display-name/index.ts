export * from './editable-tag-category-display-name.component';
