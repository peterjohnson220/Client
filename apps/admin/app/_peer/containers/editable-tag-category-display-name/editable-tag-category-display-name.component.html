<span class="editable" *ngIf="!isEditing" (click)="toggleEdit()">{{value}}</span>
<ng-container *ngIf="isEditing">
  <form [formGroup]="editDisplayNameForm" (ngSubmit)="save()">
    <div class="input-group">
      <input id="editable-category-name-input"
             type="text"
             class="form-control"
             autocomplete="off"
             formControlName="tempValue"
             pfFocus
             [pfValidatable]="true"
             [disabled]="isSaving"
             [minLength]="3">
      <div (click)="save()" class="input-group-append" [ngClass]="{'disabled': !buttonsEnabled}">
          <span class="input-group-text">
            <i class="fas fa-check-circle text-success" aria-hidden="true"></i>
          </span>
      </div>
      <div (click)="cancel()" class="input-group-append" [ngClass]="{'disabled': !buttonsEnabled}">
          <span class="input-group-text">
            <i class="fas fa-times"
               aria-hidden="true"></i>
          </span>
      </div>

    </div>

    <ng-container *ngIf="tempValueControl.errors">
      <div *ngIf="tempValueControl.errors.required" class="invalid-feedback">Display Name is required.</div>
      <div *ngIf="tempValueControl.errors.error" class="invalid-feedback">{{tempValueControl.errors.error}}</div>
    </ng-container>
  </form>
</ng-container>
