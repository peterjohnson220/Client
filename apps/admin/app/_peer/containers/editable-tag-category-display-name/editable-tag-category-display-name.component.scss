@import 'colors';

.editable {

  padding: .35rem;

  .editable-icon:not(:hover) {
    display: none;
  }

  &:hover {
    .editable-icon {
      padding-left: 1rem;
      display: inline-block;
    }
    background-color: lighten($inactive-button-color, 20%);
    border-radius: 7%;
    cursor: pointer;
  }
}

.disabled {
  opacity: .5;
}
