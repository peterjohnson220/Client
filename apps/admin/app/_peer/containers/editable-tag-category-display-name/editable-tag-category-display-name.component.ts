import {Component, EventEmitter, Input, OnDestroy, OnInit, Output} from '@angular/core';
import {FormBuilder, FormGroup} from '@angular/forms';

import { Observable, Subscription } from 'rxjs';

import { PfValidators } from 'libs/forms/validators';

@Component({
  selector: 'pf-editable-tag-category-display-name',
  templateUrl: './editable-tag-category-display-name.component.html',
  styleUrls: ['./editable-tag-category-display-name.component.scss']
})
// TODO: WRITE UNIT TESTS
export class EditableTagCategoryDisplayNameComponent implements OnInit, OnDestroy {
  @Input() value: string;
  @Input() saving$: Observable<boolean>;
  @Input() errorSaving$: Observable<boolean>;
  @Output() onSave = new EventEmitter();

  savingSubscription: Subscription;
  editDisplayNameForm: FormGroup;
  isEditing: boolean;
  isSaving: boolean;

  constructor(private fb: FormBuilder) {
    this.createForm();
  }

  get tempValueControl() { return this.editDisplayNameForm.get('tempValue'); }

  get isDirty(): boolean { return this.value !== this.tempValueControl.value; }

  get buttonsEnabled(): boolean {
    return this.isDirty && !this.isSaving && this.tempValueControl.valid;
  }

  createForm(): void {
    this.editDisplayNameForm = this.fb.group({
      'tempValue': [this.value, [PfValidators.required, PfValidators.minLengthTrimWhitespace(3)]]
    });
  }

  save(): void {
    if (!this.buttonsEnabled) {
      return;
    }

    this.onSave.emit(this.tempValueControl.value);
  }

  cancel(): void {
    if (this.isSaving) {
      return;
    }

    this.reset();
  }

  reset(): void {
    this.isEditing = false;
    this.tempValueControl.setValue(this.value);
  }

  toggleEdit(): void {
    this.isEditing = true;
    this.tempValueControl.setValue(this.value);
  }

  // Events
  ngOnInit(): void {
    this.savingSubscription = this.saving$.subscribe(saving => {
      this.isSaving = saving;
      if (!saving && this.isEditing) {
        this.reset();
      }
    });
  }

  ngOnDestroy(): void {
    this.savingSubscription.unsubscribe();
  }
}
