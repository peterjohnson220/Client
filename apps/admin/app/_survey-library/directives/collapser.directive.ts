import {Directive, ElementRef, HostListener, Input} from '@angular/core';

@Directive({
  selector: '[pfCollapser]'
})
export class CollapserDirective {

  @Input('pfCollapser') surveyTitleId: number;
  public collapsed: boolean;

  constructor(private el: ElementRef) { this.collapsed = true; }

  @HostListener('click') onCategoryClick() {
    this.el.nativeElement.className += ' active';
    const element = document.querySelector('#survey' + this.surveyTitleId);
    if (this.collapsed) {
      element.classList.add('show');
    } else {
      element.classList.remove('show');
    }
    this.collapsed = !this.collapsed;
  }
}
