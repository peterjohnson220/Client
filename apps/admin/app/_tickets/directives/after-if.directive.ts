import { AfterContentInit, Directive, EventEmitter, Output } from '@angular/core';

// tslint:disable-next-line:directive-selector
@Directive({selector: '[pf-afterIf]'})
export class AfterIfDirective implements AfterContentInit {
  @Output('pf-afterIf')
  public after: EventEmitter<AfterIfDirective> = new EventEmitter();

  public ngAfterContentInit(): void {
    setTimeout(() => {
      // timeout helps prevent unexpected change errors
      this.after.next(this);
    });
  }
}
