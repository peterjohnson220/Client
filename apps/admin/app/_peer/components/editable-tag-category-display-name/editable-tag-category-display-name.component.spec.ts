import { NO_ERRORS_SCHEMA } from '@angular/core';
import { TestBed, ComponentFixture } from '@angular/core/testing';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';

import { of } from 'rxjs/internal/observable/of';

import { PfCommonModule } from 'libs/core';
import { PfValidatableDirective } from 'libs/forms/directives';

import { EditableTagCategoryDisplayNameComponent } from './editable-tag-category-display-name.component';

describe('Admin - Peer - Editable Tag Category Display Name', () => {
  let fixture: ComponentFixture<EditableTagCategoryDisplayNameComponent>;
  let instance: EditableTagCategoryDisplayNameComponent;

  // Configure Testing Module for before each test
  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        ReactiveFormsModule,
        FormsModule,
        PfCommonModule
      ],
      declarations: [
        PfValidatableDirective,
        EditableTagCategoryDisplayNameComponent
      ],
      // Shallow Testing
      schemas: [ NO_ERRORS_SCHEMA ]
    });

    fixture = TestBed.createComponent(EditableTagCategoryDisplayNameComponent);
    instance = fixture.componentInstance;

    instance.value = 'MOCK_VALUE';
    instance.saving$ = of(false);
    instance.errorSaving$ = of(false);
  });

  it('should match snapshot', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it(`should show edit form when toggleEdit is triggered`, () => {
    fixture.detectChanges();

    instance.toggleEdit();

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();


  });
});
