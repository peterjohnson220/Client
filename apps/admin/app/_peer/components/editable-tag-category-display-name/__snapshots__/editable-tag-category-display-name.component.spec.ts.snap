// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin - Peer - Editable Tag Category Display Name should match snapshot 1`] = `
<pf-editable-tag-category-display-name
  editDisplayNameForm={[Function FormGroup]}
  errorSaving$={[Function Observable]}
  fb={[Function FormBuilder]}
  isSaving="false"
  onSave={[Function EventEmitter_]}
  saving$={[Function Observable]}
  savingSubscription={[Function Subscriber]}
  value={[Function String]}
>
  <span
    class="editable"
    title="Click to Edit Display Name"
  >
     MOCK_VALUE 
    <span
      class="editable-icon"
    >
      <fa-icon
        icon="pencil"
      />
    </span>
  </span>
</pf-editable-tag-category-display-name>
`;

exports[`Admin - Peer - Editable Tag Category Display Name should show edit form when toggleEdit is triggered 1`] = `
<pf-editable-tag-category-display-name
  editDisplayNameForm={[Function FormGroup]}
  errorSaving$={[Function Observable]}
  fb={[Function FormBuilder]}
  isEditing={[Function Boolean]}
  isSaving="false"
  onSave={[Function EventEmitter_]}
  saving$={[Function Observable]}
  savingSubscription={[Function Subscriber]}
  value={[Function String]}
>
  <form
    class="ng-untouched ng-pristine ng-valid"
    novalidate=""
  >
    <div
      class="input-group"
    >
      <input
        autocomplete="off"
        class="form-control ng-untouched ng-pristine ng-valid"
        formcontrolname="tempValue"
        id="editable-category-name-input"
        minlength="3"
        pffocus=""
        type="text"
      />
      <div
        class="input-group-append disabled"
      >
        <span
          class="input-group-text"
        >
          <fa-icon
            class="text-success"
            icon="check-circle"
          />
        </span>
      </div>
      <div
        class="input-group-append"
      >
        <span
          class="input-group-text"
        >
          <fa-icon
            icon="times"
          />
        </span>
      </div>
    </div>
    
  </form>
</pf-editable-tag-category-display-name>
`;
